runtime/js/40_command.js

```diff
@@ -0,0 +1,198 @@
+    cwd = undefined,
+    clearEnv = false,
+    env = {},
+    uid = undefined,
+    gid = undefined,
+    stdin = "null",
+    stdout = "inherit",
+    stderr = "inherit",
+  } = {}) {
+    const child = core.opSync("op_command_spawn", {
+      cmd: pathFromURL(command),
+      args: ArrayPrototypeMap(args, String),
+      cwd: pathFromURL(cwd),
+      clearEnv,
+      env: ObjectEntries(env),
+      uid,
+      gid,
+      stdin,
+      stdout,
+      stderr,
+    });
+    return new Child(illegalConstructorKey, child);
+  }
+
+  class Child {
+    #rid;
+
+    #pid;
+    get pid() {
+      return this.#pid;
+    }
+
+    #stdinRid;
+    #stdin = null;
+    get stdin() {
+      return this.#stdin;
+    }
+
+    #stdoutRid;
+    #stdout = null;
+    get stdout() {
+      return this.#stdout;
+    }
+
+    #stderrRid;
+    #stderr = null;
+    get stderr() {
+      return this.#stderr;
+    }
+
+    constructor(key = null, {
+      rid,
+      pid,
+      stdinRid,
+      stdoutRid,
+      stderrRid,
+    } = null) {
+      if (key !== illegalConstructorKey) {
+        throw new TypeError("Illegal constructor.");
+      }
+
+      this.#rid = rid;
+      this.#pid = pid;
+
+      if (stdinRid !== null) {
+        this.#stdinRid = stdinRid;
+        this.#stdin = writableStreamForRid(stdinRid);
+      }
+
+      if (stdoutRid !== null) {
+        this.#stdoutRid = stdoutRid;
+        this.#stdout = readableStreamForRid(stdoutRid);
+      }
+
+      if (stderrRid !== null) {
+        this.#stderrRid = stderrRid;
+        this.#stderr = readableStreamForRid(stderrRid);
+      }
+    }
+
+    #status;
+    get status() {
+      if (this.#status) {
+        return this.#status;
+      }
+      const status = core.opSync("op_command_status", this.#rid);
+      // TODO(@crowlKats): 2.0 change typings to return null instead of undefined for status.signal
+      if (status) {
+        status.signal ??= undefined;
+      }
+      return status;
+    }
+
+    async wait() {
+      if (this.#rid === null) {
+        throw new TypeError("Child process has already terminated.");
+      }
+      const status = await core.opAsync(
+        "op_command_wait",
+        this.#rid,
+        this.#stdinRid,
+      );
+      await this.stdin?.abort();
+      this.#rid = null;
+      // TODO(@crowlKats): 2.0 change typings to return null instead of undefined for status.signal
+      status.signal ??= undefined;
+      this.#status = status;
+      return status;
+    }
+
+    async output() {
+      if (this.#rid === null) {
+        throw new TypeError("Child process has already terminated.");
+      }
+      const res = await core.opAsync("op_command_output", {
+        rid: this.#rid,
+        stdoutRid: this.#stdoutRid,
+        stderrRid: this.#stderrRid,
+      });
+      await this.stdin?.abort();
+      this.#rid = null;
+      // TODO(@crowlKats): 2.0 change typings to return null instead of undefined for status.signal
+      res.status.signal ??= undefined;
```

Comment:
This could be done now?

Reply:
@bartlomieju we could if we have a new interface and remove the old one, but cant change the existing one without breaking `Deno.run`. I dont really care eitherway, you tell me

Reply:
Ah I see, do you think we could introduce a separate interface so there are no changes to `Deno.run` whatsoever?

Reply:
sure